File: 129-sum-root-to-leaf-numbers/129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void DFS(TreeNode* root,string &str, int &res)
    {   
        str.push_back(char(root->val + 48));
        
        if(root->left==NULL && root->right==NULL)
        {
            res = res + stoi(str);
            str.pop_back();
            return;
        }
        
         if(root->left!=NULL)
            DFS(root->left,str,res);
         if(root->right!=NULL)
             DFS(root->right,str,res);
        
        str.pop_back();
        
        return; 
    }
    
    int sumNumbers(TreeNode* root) {
        
        int res = 0;
        
        string str = "";
        
        DFS(root,str,res);
        
        return res;
    }
};

File: 129-sum-root-to-leaf-numbers/129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
def traverse(A,sum,k):
    p=k[:]
    p+=str(A.val)
    if A.left==None and A.right==None:
        sum[0]+=int(p)
    if A.left:
        traverse(A.left,sum,p)
    if A.right:
        traverse(A.right,sum,p)
    
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        sum=[0]
        traverse(root,sum,"")
        return sum[0]